lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -2134,37 +2134,51 @@ namespace Js
     Var JavascriptString::ToCaseCore(JavascriptString* pThis, ToCase toCase)
     {
         charcount_t count = pThis->GetLength();
+        charcount_t countToCase = count;
 
+        const wchar_t* inStr = pThis->GetString();
+        const wchar_t* inStrLim = inStr + count;
+        const wchar_t* i = inStr;
+
+        // Try to find out the chars that do not need casing
+        if (toCase == ToUpper)
+        {
+            while (i < inStrLim) {
+                if (*i >= 224 || (*i <= 118 && *i >= 97)) break;
+                i++;
+            }
+        } else {
+            Assert(toCase == ToLower);
+            while (i < inStrLim) {
+                if (*i >= 257 || (*i <= 204 && *i >= 192) || (*i <= 90 && *i >= 65)) break;
+                i++;
+            }
+        }
+
+        // If no char needs casing, return immediately
+        if (i == inStrLim) return pThis;
+
+        // Otherwise, copy the string and start casing
+        countToCase = (charcount_t)(inStrLim - i);
         BufferStringBuilder builder(count, pThis->type->GetScriptContext());
-        const wchar_t *inStr = pThis->GetString();
-        wchar_t *outStr = builder.DangerousGetWritableBuffer();
+        wchar_t* outStr = builder.DangerousGetWritableBuffer();
 
         wchar_t* outStrLim = outStr + count;
-        wchar_t *o = outStr;
+        wchar_t* o = outStr;
 
         while (o < outStrLim)
         {
             *o++ = *inStr++;
         }
```

Comment:
I wonder why we didn't use wmemcpy here.  @abchatra or @curtisman any idea why we wouldn't use wmemcpy?